### PR TITLE
Added UI Component to DG

### PR DESCRIPTION
**Pull request by heekit73098**

### CODE REVIEW

File: docs/UI.puml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@startuml
+Class TextUI{
+    + getUserCommand()
+    + showMessage(message: Object)
+}
+Scanner -* TextUI
+PrintStream <-- TextUI
```
Comment (ngys117) [inline docs/UI.puml]: Should PrintStream be considered as a composition as well if Scanner is? PrintStream and Scanner should not exist if TextUi does not exist is what I'm understanding.

Comment (heekit73098) [reply docs/UI.puml]: I did it like that because i thought the PrintSream Object is System.out. So it should exists even when TextUi is destroyed

Comment (Ch40gRv1-Mu) [reply docs/UI.puml]: I may support Yongsheng's point of view. The reason is that all interactive I/O(ui) is done by the ui held by Main, which means the UI cannot be a independent component out of Main? (The System.out is just a API called by PrintSream Object but not PrintSream Object itself, if we wanna make a GUI can directly change the System.out to other output format and our App will still run as expected)

Comment (heekit73098) [reply docs/UI.puml]: Removed the Class Diagram for now instead to see how we can improve on it in the future.

**Comment by chooyikai** [comment]
Did you remember to merge the existing DG file into your own properly? There seems to be some duplicate segments and stuff

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1,5 +1,25 @@
 # Developer Guide
 
+## Design
+### Architecture
+Given below is a quick overview of the main components of Mod Happy and how they interact with one another.
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/Ch40gRv1-Mu/tp/branch-A-UML/docs/Components.puml)
+
+
```
Comment (chooyikai) [inline docs/DeveloperGuide.md]: Seems like this section was duplicated during the merge (see lines 28-34 of the same file)

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+It does not interact with any other classes or components and serves strictly as the gateway for the
+user to interact with the application.
+<br>
+The `TextUi` component:
+- Listens and grabs the user's input from the standard input using an `Java.util.Scanner` object.
+- Displays any command results and error messages on the standard output using the built-in `Java.io.PrintStream` object `Java.System.out`.
+
+### Parser Component
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/Ch40gRv1-Mu/tp/branch-A-UML/docs/Parser.puml)
+
```
Comment (chooyikai) [inline docs/DeveloperGuide.md]: Same, duplicated section as well

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1,5 +1,25 @@
 # Developer Guide
 
+## Design
+### Architecture
+Given below is a quick overview of the main components of Mod Happy and how they interact with one another.
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/Ch40gRv1-Mu/tp/branch-A-UML/docs/Components.puml)
+
+
+### UI Component
```
Comment (chooyikai) [inline docs/DeveloperGuide.md:13]: This section can probably just be slotted in with the other components below (lines 33+)

**Pull request merged**